### ISSUE TITLE
go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.41.1 complains about gocognit v1.0.1

### ISSUE THREAD

**Issue by pawel-kedzior-sw** [golangci/golangci-lint#2130]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [ ] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

When I issue:
go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.41.1

I receive this complain:
```
github.com/golangci/golangci-lint/cmd/golangci-lint imports
        github.com/golangci/golangci-lint/pkg/commands imports
        github.com/golangci/golangci-lint/pkg/lint/lintersdb imports
        github.com/golangci/golangci-lint/pkg/golinters imports
        github.com/uudashr/gocognit: reading github.com/uudashr/gocognit/go.mod at revision v1.0.1: unknown revision v1.0.1
```

golangci-lint does not get installed

### Version of golangci-lint

v1.41.1

### Configuration file

N/A

### Go environment

<details>

```console
go version go1.16.5 windows/amd64
set GO111MODULE=on
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\foo.bar\AppData\Local\go-build
set GOENV=C:\Users\foo.bar\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GOMODCACHE=C:\Users\foo.bar\go\pkg\mod
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=C:\Users\foo.bar\go
set GOPRIVATE=
set GOPROXY=direct
set GOROOT=C:\Program Files\Go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=C:\Program Files\Go\pkg\tool\windows_amd64
set GOVCS=
set GOVERSION=go1.16.5
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=D:\dev\qux\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=C:\Users\foo.bar\AppData\Local\Temp\go-build3788443418=/tmp/go-build -gno-record-gcc-switches
```
</details>

### Verbose output of running

<details>

```console
D:\dev\qux>go get -v github.com/golangci/golangci-lint/cmd/golangci-lint@v1.41.1
go: downloading github.com/uudashr/gocognit v1.0.1
github.com/golangci/golangci-lint/cmd/golangci-lint imports
        github.com/golangci/golangci-lint/pkg/commands imports
        github.com/golangci/golangci-lint/pkg/lint/lintersdb imports
        github.com/golangci/golangci-lint/pkg/golinters imports
        github.com/uudashr/gocognit: reading github.com/uudashr/gocognit/go.mod at revision v1.0.1: unknown revision v1.0.1
get "4d63.com/gochecknoglobals": found meta tag vcs.metaImport{Prefix:"4d63.com/gochecknoglobals", VCS:"git", RepoRoot:"https://github.com/leighmcculloch/gochecknoglobals"} at //4d63.com/gochecknoglobals?go-get=1
get "golang.org/x/sys": found meta tag vcs.metaImport{Prefix:"golang.org/x/sys", VCS:"git", RepoRoot:"https://go.googlesource.com/sys"} at //golang.org/x/sys?go-get=1
get "golang.org/x/mod": found meta tag vcs.metaImport{Prefix:"golang.org/x/mod", VCS:"git", RepoRoot:"https://go.googlesource.com/mod"} at //golang.org/x/mod?go-get=1
get "golang.org/x/text": found meta tag vcs.metaImport{Prefix:"golang.org/x/text", VCS:"git", RepoRoot:"https://go.googlesource.com/text"} at //golang.org/x/text?go-get=1
get "golang.org/x/tools": found meta tag vcs.metaImport{Prefix:"golang.org/x/tools", VCS:"git", RepoRoot:"https://go.googlesource.com/tools"} at //golang.org/x/tools?go-get=1
get "golang.org/x/xerrors": found meta tag vcs.metaImport{Prefix:"golang.org/x/xerrors", VCS:"git", RepoRoot:"https://go.googlesource.com/xerrors"} at //golang.org/x/xerrors?go-get=1
get "google.golang.org/protobuf": found meta tag vcs.metaImport{Prefix:"google.golang.org/protobuf", VCS:"git", RepoRoot:"https://go.googlesource.com/protobuf"} at //google.golang.org/protobuf?go-get=1
get "gopkg.in/ini.v1": found meta tag vcs.metaImport{Prefix:"gopkg.in/ini.v1", VCS:"git", RepoRoot:"https://gopkg.in/ini.v1"} at //gopkg.in/ini.v1?go-get=1
get "gopkg.in/yaml.v3": found meta tag vcs.metaImport{Prefix:"gopkg.in/yaml.v3", VCS:"git", RepoRoot:"https://gopkg.in/yaml.v3"} at //gopkg.in/yaml.v3?go-get=1
get "gopkg.in/yaml.v2": found meta tag vcs.metaImport{Prefix:"gopkg.in/yaml.v2", VCS:"git", RepoRoot:"https://gopkg.in/yaml.v2"} at //gopkg.in/yaml.v2?go-get=1
get "honnef.co/go/tools": found meta tag vcs.metaImport{Prefix:"honnef.co/go/tools", VCS:"git", RepoRoot:"https://github.com/dominikh/go-tools"} at //honnef.co/go/tools?go-get=1
get "mvdan.cc/gofumpt": found meta tag vcs.metaImport{Prefix:"mvdan.cc/gofumpt", VCS:"git", RepoRoot:"https://github.com/mvdan/gofumpt"} at //mvdan.cc/gofumpt?go-get=1
get "mvdan.cc/lint": found meta tag vcs.metaImport{Prefix:"mvdan.cc/lint", VCS:"git", RepoRoot:"https://github.com/mvdan/lint"} at //mvdan.cc/lint?go-get=1
get "mvdan.cc/unparam": found meta tag vcs.metaImport{Prefix:"mvdan.cc/unparam", VCS:"git", RepoRoot:"https://github.com/mvdan/unparam"} at //mvdan.cc/unparam?go-get=1
get "mvdan.cc/interfacer": found meta tag vcs.metaImport{Prefix:"mvdan.cc/interfacer", VCS:"git", RepoRoot:"https://github.com/mvdan/interfacer"} at //mvdan.cc/interfacer?go-get=1
```
</details>

### Code example or link to a public repository

<details>
It seems that there is v1.0.1 of gocognit:
https://github.com/uudashr/gocognit/tree/v1.0.1
</details>

**Comment by ldez** [comment]
Hello,

the version `v1.0.1` exists, can you check your go proxy?

https://github.com/uudashr/gocognit/releases/tag/v1.0.1

**Comment by pawel-kedzior-sw** [comment]
@ldez - thank you, this is good direction
my `GOPROXY` was set to `direct` when the issue was present,
after setting:
`set GOPROXY=https://proxy.golang.org`
`go get` went smoothly.

This solves problem for me, but remains peculiar in general, why go complains about the direct download.

**Comment by ldez** [comment]
I think we can close this issue.
